### PR TITLE
bsc#1193212: use the ProductSpec API

### PR DESCRIPTION
**Pull request by imobachgs**
Tentative fix for [bsc#1193212](https://bugzilla.suse.com/show_bug.cgi?id=1193212).

Related PRs:

* https://github.com/yast/yast-packager/pull/595
* https://github.com/yast/yast-yast2/pull/1207

### CODE REVIEW

File: src/include/update/rootpart.rb [hunk 1]
```diff
@@ -523,7 +523,7 @@ def RootPartitionDialog(flavor)
 
     def target_distribution
       # FIXME: this is the same as in src/lib/update/clients/inst_update_partition_auto.rb:113
-      if Y2Packager::MediumType.online?
+      if Y2Packager::MediumType.offline? || Y2Packager::MediumType.online?
```
[the remark this answers is not in the document]
Comment (imobachgs) [reply src/include/update/rootpart.rb:526]: I do not like to rely on `MediumType` at all (we removed several references to those methods during the refactoring). I am trying to find a better (and cleaner) solution.

**Comment by imobachgs** [comment]
I have updated the code to do not rely on `MediumType`, but I needed to adapt other packages too:

* https://github.com/yast/yast-packager/pull/595
* https://github.com/yast/yast-yast2/pull/1207

I am running some tests before considering the problem as fixed.

**Review by jreidinger — APPROVED** [review]
change loooks good. Just code is still identical on both places.

**Comment by imobachgs** [comment]
> change loooks good. Just code is still identical on both places.

Yes, you are right. At least, they are just two lines now.

**Review by jreidinger — APPROVED** [review]

**Pull request merged**